File: example/model/base_model.go
package model

import (
	"time"
)

type BaseModel struct {
	ID        int64      `json:"id"`
	UUID      string     `json:"uuid"`
	CreatedAt time.Time  `json:"created_at"`
	UpdatedAt time.Time  `json:"updated_at"`
	DeletedAt *time.Time `json:"deleted_at"`
}

File: cmd/v1/service/generator_v2/test/app_ts/domain/model/base_model.ts
export class BaseModel {
    ID: bigint;
    UUID: string;
    createdAt: string;
    createdBy: string;
    updatedAt: string;
    updatedBy: string;
    deletedAt: string;
    deletedBy: string;

    constructor(
        ID: bigint,
        UUID: string,
        createdAt: string,
        createdBy: string,
        updatedAt: string,
        updatedBy: string,
        deletedAt: string,
        deletedBy: string
    ) {
        this.ID = ID;
        this.UUID = UUID;
        this.createdAt = createdAt;
        this.createdBy = createdBy;
        this.updatedAt = updatedAt;
        this.updatedBy = updatedBy;
        this.deletedAt = deletedAt;
        this.deletedBy = deletedBy;
    }
}
